unix/mpconfigport.h

```diff
@@ -250,3 +250,8 @@ extern const struct _mp_obj_fun_builtin_t mp_builtin_open_obj;
 #include <alloca.h>
 #endif
 #endif
+
+#ifndef _DIRENT_HAVE_D_TYPE
```

Comment:
From man: "Under  glibc,  programs  can  check  for the availability of the fields not defined in POSIX.1 by testing whether the       macros _DIRENT_HAVE_D_NAMLEN, _DIRENT_HAVE_D_RECLEN, _DIRENT_HAVE_D_OFF, or _DIRENT_HAVE_D_TYPE are defined."

musl for example doesn't define these.